src/Framework/PaymentGateways/Routes/GatewayRouteFacade.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Give\Framework\PaymentGateways\Routes;
+
+use Give\Framework\PaymentGateways\DataTransferObjects\GatewayOffsiteReturnData;
+use Give\Framework\PaymentGateways\Exceptions\PaymentGatewayException;
+use Give\Framework\PaymentGateways\PaymentGateway;
+use Give\Framework\PaymentGateways\PaymentGatewayRegister;
+
+/**
+ * @unreleased
+ */
+class GatewayRouteFacade
+{
+    /**
+     * @var array
+     */
+    protected $paymentGateways = [];
+    /**
+     * @var PaymentGatewayRegister
+     */
+    private $paymentGatewayRegister;
+
+    /**
+     * @param  PaymentGatewayRegister  $paymentGatewayRegister
+     */
+    public function __construct(PaymentGatewayRegister $paymentGatewayRegister)
+    {
+        $this->paymentGatewayRegister = $paymentGatewayRegister;
+        $this->paymentGateways = $this->paymentGatewayRegister->getPaymentGateways();
```

Comment:
I don't recommend doing this as it's splitting the source of truth. The gateways are locked in at this point, here. But if gateways are modified before being used within this class, it won't be reflected.

Reply:
Are you talking about the paymentGateways property?